File: 프로그래머스/1/388351. 유연근무제/유연근무제.js
function solution(schedules, timelogs, startday) {
    let gift = 0;
    const n = schedules.length;
    
    for (let idx = 0; idx < n; idx++) {
        const goal = schedules[idx];
        
        // 출근 마감 시간 계산 (goal + 10분)
        const goalHour = Math.floor(goal / 100);
        const goalMinute = goal % 100;
        
        let deadlineMinute = goalMinute + 10;
        let deadlineHour = goalHour + Math.floor(deadlineMinute / 60);
        deadlineMinute %= 60;
        
        const deadline = deadlineHour * 100 + deadlineMinute;
        
        // 출근 상태 확인
        let onTimeCount = 0;
        let neededDays = 0;
        let currentDay = startday;
        
        for (let i = 0; i < 7; i++) {
            // 주말인 경우 건너뛰기
            if (currentDay === 6 || currentDay === 7) {
                currentDay = currentDay === 7 ? 1 : currentDay + 1;
                continue;
            }
            
            // 평일인 경우 출근 체크
            neededDays++;
            if (timelogs[idx][i] <= deadline) {
                onTimeCount++;
            }
            
            // 평일 5일을 모두 확인했으면 종료
            if (neededDays >= 5) {
                break;
            }
                
            // 다음 날로 이동
            currentDay = currentDay === 7 ? 1 : currentDay + 1;
        }
        
        // 모든 평일에 지각하지 않았는지 확인
        if (onTimeCount === neededDays) {
            gift++;
        }
    }
    
    return gift;
}

File: 프로그래머스/1/388351. 유연근무제/유연근무제.py
def solution(schedules, timelogs, startday):
    gift = 0
    n = len(schedules)
    
    for idx, goal in enumerate(schedules):
        cnt = 0
        week = startday
        workdays = 0
        
        # Calculate deadline time (goal + 10 minutes)
        goal_hour = goal // 100
        goal_minute = goal % 100
        
        # Add 10 minutes and handle hour overflow
        deadline_minute = goal_minute + 10
        deadline_hour = goal_hour + (deadline_minute // 60)
        deadline_minute %= 60
        
        deadline = deadline_hour * 100 + deadline_minute
        
        for i in range(7):
            if week in [6, 7]:  # Weekend
                week = 1 if week == 7 else week + 1
                continue
            
            # Check if employee arrived on time
            if timelogs[idx][i] <= deadline:
                cnt += 1
            
            workdays += 1
            
            if workdays >= 5:  # We've checked all weekdays
                break
            
            week = 1 if week == 7 else week + 1  
            
        if cnt == workdays:  # All weekdays were attended on time
            gift += 1
    
    return gift